pyaedt/circuit.py

```diff
@@ -1687,3 +1688,404 @@ def import_edb_in_circuit(self, edb_path):
         hfss_3d_layout_model = self.modeler.schematic.add_subcircuit_3dlayout(hfss.design_name)
         hfss.close_project(save_project=False)
         return hfss_3d_layout_model
+
+    @pyaedt_function_handler()
+    def create_tdr_schematic_from_snp(
+        self,
+        touchstone,
+        probe_p_pins,
+        probe_ref_pins=None,
+        termination_pins=None,
+        differential=True,
+        rise_time=30,
+        use_convolution=True,
+        analyze=True,
+        design_name="LNA",
+    ):
+        """Create a new schematic from a touchstone file and automatically setup a TDR transient analysis.
+
+        Parameters
+        ----------
+        touchstone : str
+            Full path to the sNp file.
+        probe_p_pins : list
+            Pins to attach to the probe components.
+        probe_ref_pins : list, optional
+            Reference pins to attach to probe components. Valid only in differentila con
```

Comment:
```suggestion
            Reference pins to attach to probe components. The default is ``None``.
            This parameter is valid only in differential con
```
The last sentence is incomplete.